Update dto definitions

joennlae opened the pull request:
First of all :-) Fantastic project :100 Thank you!

### Checklist

* [x] I have read the Contribution Guidelines
* [x] I have read and agree to the Code of Conduct
* [x] I have read and agree that my contributions will be included under the project license
* [x] I have added a description of my changes and why I want them _or_ linked the relevant pull request below
* [x] I have ran the formatter, linter and tests locally and ensured they give their all green
* [x] If contributing new code: I've ensured my new code is covered sufficiently by unit and integration tests

### Description of Changes

This changes the dto field names to conform to the current definition: https://github.com/pocketbase/js-sdk/blob/master/src/tools/dtos.ts#L136

Additionally, this adds some fields used in `pocketbase_orm`. These are all fields delivered from the official collections API.

https://github.com/44ai-labs/pocketbase_orm/tree/update-reference-mechanism

thijsmie commented:
Hi there! Some good changes here, but I don't see any additions to the collectionfield in the official js sdk: https://github.com/pocketbase/js-sdk/blob/b17fd45624a86d69b56163da6e8a519f533a1f22/src/tools/dtos.ts#L33. Pocketbase_orm seems to be an unrelated project, that should not affect the types used in the SDK right?

joennlae commented:
Hi @thijsmie 

You are right. I am just integrating and updating this project: https://github.com/44ai-labs/pocketbase_orm/tree/update-reference-mechanism

That's why I added the comments. We can also leave the undefined fields as with `TypedDict` that you use, we can do that. The `vaphnes` library uses `pydantic` so we had to define these fields.

I updated it.
